relayer-cli/src/commands/query/channel_ends.rs

```diff
@@ -0,0 +1,182 @@
+use std::sync::Arc;
+
+use abscissa_core::{Command, Options, Runnable};
+use serde::{Deserialize, Serialize};
+use tokio::runtime::Runtime as TokioRuntime;
+
+use ibc::ics03_connection::connection::ConnectionEnd;
+use ibc::ics04_channel::channel::ChannelEnd;
+use ibc::ics07_tendermint::client_state::ClientState;
+use ibc::ics24_host::identifier::ChainId;
+use ibc::ics24_host::identifier::{ChannelId, ClientId, ConnectionId, PortId};
+use ibc::Height;
+use ibc_relayer::chain::{Chain, CosmosSdkChain};
+
+use crate::conclude::Output;
+use crate::prelude::*;
+
+#[derive(Clone, Command, Debug, Options)]
+pub struct QueryChannelEndsCmd {
+    #[options(free, required, help = "identifier of the chain to query")]
+    chain_id: ChainId,
+
+    #[options(free, required, help = "identifier of the port to query")]
+    port_id: PortId,
+
+    #[options(free, required, help = "identifier of the channel to query")]
+    channel_id: ChannelId,
+
+    #[options(help = "height of the state to query", short = "h")]
+    height: Option<u64>,
+
+    #[options(help = "display result in summary form", short = "s")]
+    summary: bool,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TraceResult {
+    channel_end: ChannelEnd,
+    connection_end: ConnectionEnd,
+    client_state: ClientState,
+    counterparty_channel_end: ChannelEnd,
+    counterparty_connection_end: ConnectionEnd,
+    counterparty_client_state: ClientState,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TraceSummary {
+    chain_id: ChainId,
+    client_id: ClientId,
+    connection_id: ConnectionId,
+    channel_id: ChannelId,
+    port_id: PortId,
+
+    counterparty_chain_id: ChainId,
+    counterparty_client_id: ClientId,
+    counterparty_connection_id: ConnectionId,
+    counterparty_channel_id: ChannelId,
+    counterparty_port_id: PortId,
+}
+
+fn do_run(cmd: &QueryChannelEndsCmd) -> Result<(), Box<dyn std::error::Error>> {
+    debug!("Options: {:?}", cmd);
+
+    let config = app_config();
+
+    let chain_id = cmd.chain_id.clone();
+    let port_id = cmd.port_id.clone();
+    let channel_id = cmd.channel_id.clone();
+
+    let chain_config = config
+        .find_chain(&chain_id)
+        .ok_or_else(|| format!("chain '{}' not found in configuration file", chain_id))?;
+
+    let rt = Arc::new(TokioRuntime::new().unwrap());
+    let chain = CosmosSdkChain::bootstrap(chain_config.clone(), rt.clone())?;
```

Comment:
I think we can use a simpler approach here, along the lines of:

```suggestion
    let chain = spawn_chain_runtime(&config, &chain_id)?;
```

Reply:
I was copying the code from the other command implementations. Just looked at `spawn_chain_runtime` and seems like it returns a `Box<dyn ChainHandle>` which erases the type and cause some information to be unavailable. It also spawns other threads such as the event monitor, which are probably not needed here.

Reply:
I agree that the event monitor is not necessary. We can still use `CosmosSdkChain::bootstrap` if you think it's better, but it important to avoid `unwrap`s.